src/main.c

```diff
@@ -1652,7 +1683,52 @@ void se_draw_emu_stats(){
   igPopItemWidth();
 
 }
-
+#ifdef ENABLE_RETRO_ACHIEVEMENTS
+uint32_t retro_achievements_read_memory_callback(uint32_t address, uint8_t* buffer, uint32_t num_bytes, rc_client_t* client){
+  if(emu_state.system==SYSTEM_GB){
+    if (address < 0x010000U) {
+      // 0 - 0x10000 follows the normal gb memory map
+      for(int j=0;j<num_bytes;j++){
+        buffer[j]=sb_read8(&core.gb,address+j);
+      }
+    } else if (address <= 0x015FFFU) {
+      // 0x10000 - 0x15FFF is WRAM banks 2-7
+      uint8_t* wram = &core.gb.mem.wram[(SB_WRAM_BANK_SIZE * 2) + address - 0x010000U];
+      for(int j=0;j<num_bytes;j++){
+        buffer[j]=wram[j];
+      }
+    }
+    return num_bytes;
+  }else if(emu_state.system==SYSTEM_GBA){
+    const rc_memory_regions_t* regions = rc_console_memory_regions(RC_CONSOLE_GAMEBOY_ADVANCE);
+    for (int i=0;i<regions->num_regions;i++) {
+      const rc_memory_region_t* region = &regions->region[i];
+      if (address >= region->start_address && address < region->end_address) {
```

Comment:
`address >= region->start_address && address <= region->end_address` as the range is inclusive